metadata-io/src/main/java/com/linkedin/metadata/graph/neo4j/Neo4jGraphService.java

```diff
@@ -314,22 +473,86 @@ public void removeEdgesFromNode(
     // also delete any relationship going to or from it
     final RelationshipDirection relationshipDirection = relationshipFilter.getDirection();
 
-    String matchTemplate = "MATCH (src {urn: $urn})-[r%s]-(dest) DELETE r";
-    if (relationshipDirection == RelationshipDirection.INCOMING) {
-      matchTemplate = "MATCH (src {urn: $urn})<-[r%s]-(dest) DELETE r";
-    } else if (relationshipDirection == RelationshipDirection.OUTGOING) {
-      matchTemplate = "MATCH (src {urn: $urn})-[r%s]->(dest) DELETE r";
+    String deleteTemplateIndirect = "MATCH (src {urn: $urn})<-[r%s]-(dest) DELETE r";
+    String deleteTemplateDirect = "MATCH (src {urn: $urn})-[r%s]->(dest) DELETE r";
+
+    if (relationshipDirection != RelationshipDirection.INCOMING
+        && relationshipDirection != RelationshipDirection.OUTGOING) {
+      String matchTemplate = "MATCH (src {urn: $urn})-[r%s]-(dest) DELETE r";
+      String relationshipTypeFilter = "";
+      if (relationshipTypes.size() > 0) {
+        relationshipTypeFilter =
+            ":" + StringUtils.join(relationshipTypes, "|") + "|r_" + StringUtils.join(relationshipTypes, "|r_");
+      }
+      runRemoveEdgesFromNodeQuery(matchTemplate, relationshipTypeFilter, urn);
+    } else if (relationshipTypes.size() > 0) {
+      if (relationshipDirection == RelationshipDirection.INCOMING) {
+        String matchTemplate = "MATCH (src {urn: '%s'})<-[r]-(dest) RETURN DISTINCT type(r)";
+        runMatchRelationshipTypesRemove(matchTemplate, deleteTemplateIndirect, deleteTemplateDirect, urn,
+            relationshipTypes);
+      } else {
+        String matchTemplate = "MATCH (src {urn: '%s'})-[r]->(dest) RETURN DISTINCT type(r)";
+        runMatchRelationshipTypesRemove(matchTemplate, deleteTemplateDirect, deleteTemplateIndirect, urn,
+            relationshipTypes);
+      }
+    } else {
+      if (relationshipDirection == RelationshipDirection.INCOMING) {
+        String matchTemplate = "MATCH (src {urn: '%s'})<-[r]-(dest) RETURN DISTINCT type(r)";
+        runMatchRelationshipTypesRemove(matchTemplate, deleteTemplateIndirect, deleteTemplateDirect, urn, null);
+      } else {
+        String matchTemplate = "MATCH (src {urn: '%s'})-[r]->(dest) RETURN DISTINCT type(r)";
+        runMatchRelationshipTypesRemove(matchTemplate, deleteTemplateDirect, deleteTemplateIndirect, urn, null);
+      }
+    }
+  }
+
+  private void runMatchRelationshipTypesRemove(@Nonnull String matchTemplate, @Nonnull String finalDeleteTemplate,
```

Comment:
What does this function do? Can we find a clearer name?

Reply:
remove this function and modify removeEdgesFromNode